src/main/java/seedu/pill/util/ItemMap.java

```diff
@@ -399,6 +399,140 @@ public ItemMap getExpiringItems(LocalDate cutOffDate) {
         return expiringItems;
     }
 
+    /**
+     * Uses a specified quantity of items with the given name, consuming from the earliest expiring items first.
+     * If the quantity to use equals or exceeds an item's quantity, that item is deleted.
+     * If the quantity to use is less than an item's quantity, the item's quantity is reduced.
+     *
+     * @param itemName the name of the item to use
+     * @param quantityToUse the quantity of the item to consume
+     * @throws PillException if:
+     *         - the requested quantity exceeds the total available stock for the item
+     *         - the specified item name does not exist in the inventory
+     */
+    public void useItem(String itemName, int quantityToUse) throws PillException {
+        if (quantityToUse > this.stockCount(itemName)) {
+            LOGGER.warning("Attempt to use more items than available: name=" + itemName +
+                    ", requested=" + quantityToUse +
+                    ", available=" + this.stockCount(itemName));
```

Comment:
I like how detailed the logging is here, especially in tracking when items are completely or partially used